nixos.nix

```
@@ -90,13 +90,67 @@ in
         # Create a directory in persistent storage, so we can bind
         # mount it.
         mkDirCreationSnippet = persistentStoragePath: dir:
-          let
-            targetDir = concatPaths [ persistentStoragePath dir ];
-          in
           ''
-            if [[ ! -e "${targetDir}" ]]; then
-                mkdir -p "${targetDir}"
-            fi
+            # Given a source directory, /source, and a target directory,
+            # (/target/foo/bar/bazz, we want to "clone" the target structure
```

```suggestion
            # /target/foo/bar/bazz, we want to "clone" the target structure
```

Oops, I missed this paren.